lib/active_admin/application.rb

```diff
@@ -178,9 +178,9 @@ def namespace(name)
       name ||= :root
 
       namespace = namespaces[name] ||= begin
-        namespace = Namespace.new(self, name)
-        ActiveSupport::Notifications.publish ActiveAdmin::Namespace::RegisterEvent, namespace
-        namespace
```

Comment:
Anyways, I think it's good to keep this change in `lib` (no matter if it's good or bad) out of this PR.